CodeStarterCamp_Week3/Type.swift

```diff
@@ -7,42 +7,87 @@
 
 import Foundation
 
-struct Person {
+class Person {
     var name: String
     var money: UInt
     
-    func buySomething() {
-        print("\(self.name) buy something!")
+    init(name: String, money: UInt) {
+        self.name = name
+        self.money = money
+    }
+    
+    func buyCoffee(at coffeeShop: CoffeeShop, coffee: Coffee) {
+        guard (self.money > coffee.price) else {
+            print("잔액이 \(coffee.price - self.money)원만큼 부족합니다.")
+            return
+        }
+        
+        do {
+            try coffeeShop.order(coffee: coffee, nameOfCustomer: self.name)
+        } catch CoffeeShopError.noBaristaError { return }
+        catch {print("default ERROR")}
+    
+        self.money -= coffee.price
     }
 }
 
+
+class Barista: Person {
+    func makeCoffee(coffee: Coffee) {
+        print("\(coffee)를 만들었습니다!")
+    }
+}
+
+
 class CoffeeShop {
-    var sales: UInt
-    var barista: Person?
-    var menu: [Coffee]
-    let pickUpTable: String = "This is pick up table!"
+    private var sales: UInt = 0
+    private var menu: [Coffee] = []
+    var barista: Barista?
+    var pickUpTable: Coffee?
     
-    init() {
-        self.sales = 0
-        self.menu = [.americano(price: 4500), .cafeLatte(price: 5000)]
+    init(menu: [Coffee]) {
+        self.menu = menu
     }
```

Comment:
```suggestion
    init(menu: [Coffee] = [], sales: Int = 0) {
        self.menu = menu
        self.sales = sales
    }
```

이런식으로 수정도 가능했을 것 같아요!

Reply:
생성자 자체에서 초깃값을 주는 방븝은 생각해 보지 못한 것 같습니다.
감사합니다!